abstract/Block.js

```diff
@@ -224,15 +226,70 @@ export class Block extends BaseComponent {
     return this.$['*--cfg-done-activity'];
   }
 
-  /** @type {String} */
-  get cancelActivity() {
-    return this.$['*--cfg-cancel-activity'];
-  }
-
   get isActivityActive() {
     return this[ACTIVE_PROP];
   }
 
+  /** @param {Boolean} [force] */
+  initFlow(force = false) {
+    if (this.$['*uploadList']?.length && !force) {
+      this.set$({
+        '*currentActivity': Block.activities.UPLOAD_LIST,
+      });
+      this.setForCtxTarget('uc-modal', '*modalActive', true);
+    } else {
+      if (this.sourceList?.length === 1) {
+        // Single source case:
+        if (this.sourceList[0] === 'local') {
+          this.$['*currentActivity'] = Block.activities.UPLOAD_LIST;
+          this.openSystemDialog();
+        } else {
+          this.$['*currentActivity'] = this.sourceList[0];
```

Comment:
this won't work for external source activities because activity name != source name

Reply:
fixed